### ISSUE TITLE
Can't use ephemeral server after running "prefect server start"

### ISSUE THREAD

**Issue by ghost** [PrefectHQ/prefect#16101]
### Bug summary

In a from scratch helloworld type example, once I ran `prefect server start` once, I can no longer get the flow to run with the ephemeral server. The only fix is to delete `~/.prefect/prfiles.toml`.

### Version info

```Text
Version:             3.1.4
API version:         0.8.4
Python version:      3.12.4
Git commit:          78ee41cb
Built:               Wed, Nov 20, 2024 7:37 PM
OS/Arch:             darwin/arm64
Profile:             ephemeral
Server type:         unconfigured
Pydantic version:    2.10.1
Integrations:
  prefect-dask:      0.3.2
```

### Additional context

Steps to reproduce:
1. Delete `~/.prefect`
2. Run `prefect config view` to verify that `PREFECT_API_URL` is not set.
3. From a new venv, run a flow with the ephemeral server and verify that it succeeded. 
4. Run `touch ~/.prefect/profiles.toml`
5. Run the flow again, now it will fail with `ValueError: No Prefect API URL provided. Please set PREFECT_API_URL to the address of a running Prefect server.`

**Comment by Wh1isper** [comment]
A simple reproduct way

1. `rm -rf ~/.prefect`
2. `prefect server start`
3. `python my_workflow.py`
It will shows `ValueError: No Prefect API URL provided. Please set PREFECT_API_URL to the address of a running Prefect server.`

**Comment by Wh1isper** [comment]
`SettingContext.settings` did not include profile settings, I've create a pr to fix it: https://github.com/PrefectHQ/prefect/pull/16112, with attatch profile settings into root context

https://github.com/PrefectHQ/prefect/blob/62b100162bed359b4eede533d302125ca4c85a3f/src/prefect/context.py#L442-L464

**Comment by desertaxle** [comment]
Thanks for the issue @joe-dannce and thanks for looking into it @Wh1isper! I tracked down the root cause to be that we aren't handling empty profile files correctly. We should be falling back to the default `ephemeral` profile, but we aren't. I think this should be fixed in #16119 if either of you want to take a look!

**Comment by neoneye** [comment]
Thank you. 

The "Delete `~/.prefect`" , it solved my problem.

I was stuck in beginning of the quickstart guide. I'm a beginner.
https://docs.prefect.io/v3/get-started/quickstart#connect-to-a-prefect-api

Step A: I got `prefect server start` working on localhost, and could see the tasks being run.
Step B: I got `refect cloud login` working in the cloud, can could see the tasks being run.
Step C: I wanted to go back to `Step A`, however `prefect server start` wouldn't show anything. Running the program on localhost would crash, with `ValueError: No Prefect API URL provided. Please set PREFECT_API_URL to the address of a running Prefect server.`

I tried logout from cloud, but no luck.

Then I deleted `~/.prefect`, and now `Step C` is working. Yay.
